flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliTableauResultView.java

```diff
@@ -174,6 +180,9 @@ private void printStreamingResults(AtomicInteger receivedRowCount) {
                 case PAYLOAD:
                     List<RowData> changes = result.getPayload();
                     for (RowData change : changes) {
+                        if (Thread.currentThread().isInterrupted()) {
```

Comment:
What about moving this code block to the outer loop?

Reply:
I think it's better we leave it here. Because the payload may contains many records. If we move it outside, we can only exit after all results are printed.